clientv3/compare.go

```diff
@@ -14,5 +14,76 @@
 
 package clientv3
 
-type Compare struct {
+import (
+	pb "github.com/coreos/etcd/etcdserver/etcdserverpb"
+)
+
+type CompareTarget int
+type CompareResult int
+
+const (
+	CompareVersion CompareTarget = iota
+	CompareCreated
+	CompareModified
+	CompareValue
+)
+
+type Cmp pb.Compare
+
+func Compare(key string, t pb.Compare_CompareTarget, result string, v interface{}) Cmp {
```

Comment:
It would be nice to have string and byte operations both available to avoid extra allocations during conversion.